src/templates.hpp

```diff
@@ -15,7 +15,20 @@
 #include "macros.hpp"
 
 template <typename DTYPE, int HARDCODED_LMAX>
-inline void hardcoded_sph_template(DTYPE x, DTYPE y, DTYPE z, DTYPE x2, DTYPE y2, DTYPE z2, DTYPE *sph_i) {
+inline void hardcoded_sph_template(
+    DTYPE __attribute__((unused)) x, // these are unused in some code pathways, hopefully the compiler will compile them away
```

Comment:
I think `__attribute__((unused))` is a GCC/Clang thing, not sure it would work with Intel compilers or MSVC.

Reply:
went for `[[maybe_unused]]` which is in C++17 as `if constexpr`